Lib/test/test_ordered_dict.py

```diff
@@ -983,19 +983,19 @@ def test_popitem(self):
         self.assertEqual(c.popitem(last=True), (3, 3))
         self.assertEqual(c.counts, {'get': 0, 'set': 3, 'del': 0})
 
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
     def test_pop(self):
         c = self.type2test(3)
         for i in range(1, 4):
             c[i] = i
         self.assertEqual(c.counts, {'get': 0, 'set': 3, 'del': 0})
         self.assertEqual(c.pop(2), 2)
-        self.assertEqual(c.counts, {'get': 0, 'set': 3, 'del': 0})
+        self.assertEqual(c.counts, {'get': 1, 'set': 3, 'del': 1})
```

Comment:
we need to fix implementation, not the test.

Reply:
Might be easy to miss if you're a first time contributor but: the tests are pretty much copied from the CPython implementation and only tweaked to signify a test failing for RustPython. The tests basically test 'do we do the same thing'.

So, when something fails, we need to tweak the implementation to get in line as YunWon says.

Reply:
Hey! Thanks for the feedback - definitely something I wouldn't have noticed without the inupt :)

All totally makese sense, but clearly I'm not understanding what the implementation is supposed to be doing? Is it that `c.pop()` is not meant to update the number of `get`s and `del`s? In other words are the `get` and `del` fields only meant to be updated if we're explicitly using either the `.get("field")` or `["field"]` and `del c["field"]`?

Either way happy to make the changes just need some pointing in the right direction :)